### PR TITLE
Native french translation of Finamp

### PR DESCRIPTION
**Pull request by YiraSan**
I recently discovered Finamp and was totally amazed by how good the app is.

I wrote this text as I translated to explain the changes I made in the translations. My goal/thought are the following:

- To strike a balance between understanding and accuracy (for example: in France, no one says "URL", we simply say "adresse," even though it might be ambiguous; that's how the "average person" refers to URLs).

- In France, we don't translate much in "computing" terms. While there may be a French way of saying something, my priority is to use the most common term. For example, no one says "file de lecture" (playlist); we just say "playlist." Similarly, in France, people use the English word "weekend" to refer to the last two days of the week when we generally don't work. Another example: I don’t know anyone on Earth who uses the term "barre oblique" (slash); we just say "slash" (hence, I included the English word in French, because some people may never have seen the French term for a slash).

- A literal translation (like one done by a basic translator) would result in abominations because it doesn't capture the exact meaning. For example: "Impossible d'accéder aux données de téléchargement" for "Failed to open download DB" would literally be translated as "Echec de l’ouverture de la base de donnée de téléchargement," which is not only ugly to read but unnecessarily long. I choose words that together convey the same meaning.

In short, it has been very challenging to translate everything correctly while considering all possible use cases for short translations. I’ve tried to choose the least ambiguous options. I believe that you’ll need some change to how your translations are organized to make them relevant in more languages. I could make some examples if wanted, but I'm not totally sure if this is relevant, since I don't exactly know where each translations are used.

### CODE REVIEW

File: lib/l10n/app_fr.arb [hunk 1]
```diff
@@ -1,144 +1,134 @@
...
-  "@album": {},
-  "grid": "Mosaïque",
-  "@grid": {},
-  "sleepTimerTooltip": "Minuterie de veille",
-  "@sleepTimerTooltip": {},
-  "startupError": "Quelque chose s'est mal passé pendant le démarrage de l'application ! L'erreur en cause était : {error}\n\nVeuillez créer une issue sur Github à l'adresse github.com/UnicornsOnLSD/finamp avec une capture d'écran de cette page. Si le problème persiste, supprimez les données de l'application pour la réinitialiser.",
+  "startupError": "Une erreur est survenue lors du démarrage de l'application. L'erreur est: {error}\n\nPlease create an issue on github.com/UnicornsOnLSD/finamp with a screenshot of this page. If this problem persists you can clear your app data to reset the app.",
   "@startupError": {
-    "description": "The error message that shows when startup fails.",
+    "description": "Le message d'erreur affiché lorsque le démarrage de l'application échou.",
```
Comment (pinsarda) [inline lib/l10n/app_fr.arb:4]: échoue is missing its e

File: lib/l10n/app_fr.arb [hunk 2]
```diff
@@ -1,144 +1,134 @@
...
+  "@emptyServerUrl": {
+    "description": "Cette erreur intervient lorsque l'utilisateur tente de se connecter sans indiquer l'adresse du serveur"
+  },
+  "urlStartWithHttps": "L'adresse doit commencer par http:// ou https://",
   "@urlStartWithHttps": {
-    "description": "Error message that shows when the user submits a server URL that doesn't start with http:// or https:// (for example, ftp://0.0.0.0"
+    "description": "Cette erreur intervient lorsque l'utilisateur soumet une adresse qui ne commence pas par http:// ou https:// (par exemple, ftp://0.0.0.0)"
   },
-  "urlTrailingSlash": "L'URL ne doit pas se terminer pas une barre oblique",
+  "urlTrailingSlash": "L'adresse ne doit pas inclure de slash finale.",
```
Comment (pinsarda) [inline lib/l10n/app_fr.arb:26]: I think slash is a masculine word ?

File: lib/l10n/app_fr.arb [hunk 3]
```diff
@@ -1,144 +1,134 @@
...
+  "@communityRating": {},
+  "criticRating": "Note de la critique",
+  "@criticRating": {},
   "dateAdded": "Date d'ajout",
   "@dateAdded": {},
-  "datePlayed": "Date de dernière lecture",
+  "datePlayed": "Date de lecture",
   "@datePlayed": {},
-  "playCount": "Nombre de lectures",
+  "playCount": "nombre de lecture",
```
Comment (pinsarda) [inline lib/l10n/app_fr.arb:104]: I miss context but I'm pretty sure lecture should be plural here

File: lib/l10n/app_fr.arb [hunk 4]
```diff
@@ -1,144 +1,134 @@
...
   "@playCount": {},
+  "premiereDate": "Date de première",
+  "@premiereDate": {},
+  "productionYear": "Date de production",
+  "@productionYear": {},
+  "name": "Nom",
+  "@name": {},
   "random": "Aléatoire",
   "@random": {},
+  "revenue": "Revenue",
```
Comment (pinsarda) [inline lib/l10n/app_fr.arb:114]: I think Recettes would be more appropriate

File: lib/l10n/app_fr.arb [hunk 5]
```diff
@@ -164,69 +154,46 @@
...
+      "downloadedImages": {
+        "type": "String",
+        "example": "1 image"
       }
     }
   },
-  "dlRunning": "{count} en cours",
-  "@dlRunning": {
+  "dlComplete": "{count,plural,=1{{count} complété} other{{count} complétés}}",
+  "@dlComplete": {
```
Comment (pinsarda) [inline lib/l10n/app_fr.arb:172]: I guess it's a bad practice to mismatch the order of dlRunning dlComplete etc in French from other translations

File: lib/l10n/app_fr.arb [hunk 6]
```diff
@@ -243,9 +237,9 @@
       }
     }
   },
-  "playButtonLabel": "LECTURE",
+  "playButtonLabel": "LIRE",
   "@playButtonLabel": {},
-  "shuffleButtonLabel": "ALÉATOIRE",
+  "shuffleButtonLabel": "ALEATOIRE",
```
Comment (pinsarda) [inline lib/l10n/app_fr.arb:242]: We should choose between having accents with capital letters or not. Either ALEATOIRE and TELECHARGE or ALÉATOIRE and TÉLÉCHARGÉ. Not ALEATOIRE and TÉLÉCHARGÉ like now.

File: lib/l10n/app_fr.arb [hunk 7]
```diff
@@ -255,285 +249,339 @@
...
+  "insertedIntoQueue": "Ajouté à la file d'attente.",
+  "@insertedIntoQueue": {
+    "description": "Notification affichée lorsque l'utilisateur insère avec succès des éléments dans la file d'attente de lecture à un endroit qui n'est pas nécessairement à la fin."
+  },
   "queueReplaced": "File d'attente remplacée.",
   "@queueReplaced": {},
-  "startingInstantMix": "Démarrage du mix instantané.",
+  "removedFromPlaylist": "Retiré de la playlist.",
+  "@removedFromPlaylist": {},
+  "startingInstantMix": "Mix instanté lancé.",
```
Comment (pinsarda) [inline lib/l10n/app_fr.arb:474]: Mix instantané

File: lib/l10n/app_fr.arb [hunk 8]
```diff
@@ -255,285 +249,339 @@
...
-  "@addToMix": {},
-  "removeFromMix": "Enlever du mix",
+  "removeFromMix": "Retirer du Mix",
   "@removeFromMix": {},
-  "minutes": "Minutes",
-  "@minutes": {},
-  "redownloadedItems": "{count,plural, =0{Aucun re-téléchargement nécessaire.} =1{{count} élément re-téléchargé} other{{count} éléments re-téléchargés}}",
+  "addToMix": "Ajouter au Mix",
+  "@addToMix": {},
+  "redownloadedItems": "{count,plural, =0{Aucun re-téléchargement nécéssaire} =1{{count} élément re-téléchargé} other{{count} éléments re-téléchargés}}",
```
Comment (pinsarda) [inline lib/l10n/app_fr.arb:508]: nécessaire

File: lib/l10n/app_fr.arb [hunk 9]
```diff
@@ -255,285 +249,339 @@
...
+  "showUncensoredLogMessage": "Ce journal contient vos informations de connexion. Afficher tout de même ?",
   "resetTabs": "Réinitialiser les onglets",
-  "@resetTabs": {},
-  "noMusicLibrariesBody": "Finamp n'a trouvé aucune bibliothèque musicale. Assurez-vous que votre serveur Jellyfin contient au moins une médiathèque de type \"Musique\".",
-  "@noMusicLibrariesBody": {},
-  "noMusicLibrariesTitle": "Pas de bibliothèque musicale",
+  "noMusicLibrariesTitle": "Aucune bibliothèque musicale",
   "@noMusicLibrariesTitle": {
-    "description": "Title for message that shows on the views screen when no music libraries could be found."
+    "description": "Titre du message affiché sur l'écran de vues lorsqu'aucune bibliothèque musicale n'a été trouvée."
```
Comment (pinsarda) [inline lib/l10n/app_fr.arb:526]: lorsque aucune
(lorsqu' only works in front of pronouns, il elle on...)

File: lib/l10n/app_fr.arb [hunk 10]
```diff
@@ -255,285 +249,339 @@
...
+  "refresh": "RAFRAÎCHIR",
+  "swipeInsertQueueNext": "Lire la chanson balayée en prochain",
+  "@swipeInsertQueueNext": {},
+  "swipeInsertQueueNextSubtitle": "Activer l'insertion d'une chanson comme élément suivant dans la file d'attente lorsqu'elle est balayée dans la liste des chansons, au lieu de l'ajouter à la fin.",
+  "@swipeInsertQueueNextSubtitle": {},
+  "redesignBeta": "Essayer la bêta",
+  "@redesignBeta": {},
+  "playbackOrderShuffledTooltip": "Lecture en aléatoire. Touchez pour activer/désactiver.",
+  "@playbackOrderShuffledTooltip": {},
+  "playbackOrderLinearTooltip": "Lecture du l'ordre. Tap to toggle.",
```
Comment (pinsarda) [inline lib/l10n/app_fr.arb:538]: I don't get this line

**Review by pinsarda — COMMENTED** [review]
This translation is indeed a lot more natural for a French speaker. Apart from a few spelling mistakes I commented it sounds great. 
I am just concerned by the shuffling of the keys (for instance communityRating goes from line 2 to 96). It probably doesn't cause any problem for running the app, but I guess it would make it cumbersome to compare different translations files side by side.
Once it's a pull request for the redesign branch I'll look a bit more into it as I've only skimmed through this. 
Thanks a lot for this contribution I can't wait to use it personally !

File: lib/l10n/app_fr.arb (hunk 3 above)
Comment (YiraSan) [reply lib/l10n/app_fr.arb:104]: I also lacked context. At first, I translated it as 'lectures' which seemed sufficient. However, given the missing context, I opted for 'nombre de lectures' and for some reason, an extra 's' decided to sneak in.

File: lib/l10n/app_fr.arb (hunk 4 above)
Comment (YiraSan) [reply lib/l10n/app_fr.arb:114]: Right!

File: lib/l10n/app_fr.arb (hunk 5 above)
Comment (YiraSan) [reply lib/l10n/app_fr.arb:172]: I've just copied the app_en and translated it, as I saw there was missing field in the french translation uh

File: lib/l10n/app_fr.arb (hunk 6 above)
Comment (YiraSan) [reply lib/l10n/app_fr.arb:242]: I missed that "É", I agree.

File: lib/l10n/app_fr.arb (hunk 9 above)
Comment (YiraSan) [reply lib/l10n/app_fr.arb:526]: I've made those changes at night so...
Sorry for those basic errors !!

File: lib/l10n/app_fr.arb (hunk 10 above)
Comment (YiraSan) [reply lib/l10n/app_fr.arb:538]: En français y'a vraiment une difficulté concernant l'aléatoire, je n'aime pas vraiment utilisé les termes "mélanger" car en soit on parle juste de "lecture aléatoire", cette ligne si j'ai bien compris, est censé lire dans l'ordre, je m'étais trompé dans l'écriture, je pense changer comme ça dans mon commit de correction à venir :
```suggestion
  "playbackOrderLinearTooltip": "Lecture dans l'ordre. Touchez pour activer/désactiver.",
```

File: lib/l10n/app_fr.arb (hunk 9 above)
Comment (pinsarda) [reply lib/l10n/app_fr.arb:526]: No prob, don't take the very short comments as reproaches. After the first few, I saved some time and went straight to the point.
And again thanks, you did a great job !

File: lib/l10n/app_fr.arb (hunk 10 above)
Comment (pinsarda) [reply lib/l10n/app_fr.arb:538]: À voir dans son contexte effectivement, mais ça me semble bon 👌

**Comment by Chaphasilor** [comment]
@YiraSan it's fine, we can merge your changes to the `main` branch for now. I need to merge the translation updates from there into the redesign branch anyway.  
If you want to make changes to the redesign branch, please *don't* use Weblate for now, because Weblate only works with the `main` branch at the moment. Just make another PR in that case :)

Thanks for your contribution!
